runtime/compiler/ilgen/Walker.cpp

```diff
@@ -3512,8 +3513,59 @@ TR_J9ByteCodeIlGenerator::genInvokeWithVFTChild(TR::SymbolReference *symRef)
    return genInvoke(symRef, vftLoad);
    }
 
+/** @brief
+  *   Tries to find the corresponding symbol reference for the field `value` that is returned by the implementations of
+  *   BOXED_TYPE.TYPEValue() method based  on the recognized method.
+  *   If the method is not one of the recognized getter methods, this method has a fatal assert.
+  *
+  * @param generator
+  *   The current IL generator instance.
+  * @param method
+  *   The recognized method, one of the getters belonging to a primitive boxed class.
+  * @return
+  *   The symbol reference for the field holding the primitive value.
+  */
+static TR::SymbolReference * getPrimitiveValueFieldSymbolReference(TR_J9ByteCodeIlGenerator * generator, TR::RecognizedMethod method)
+   {
+   const char * className = NULL;
+   switch (method)
+      {
+      case TR::java_lang_Boolean_booleanValue:
+         className = "Ljava/lang/Boolean;";
+         break;
+      case TR::java_lang_Byte_byteValue:
+         className = "Ljava/lang/Byte;";
+         break;
+      case TR::java_lang_Character_charValue:
+         className = "Ljava/lang/Character;";
+         break;
+      case TR::java_lang_Short_shortValue:
+         className = "Ljava/lang/Short;";
+         break;
+      case TR::java_lang_Integer_intValue:
+         className = "Ljava/lang/Integer;";
+         break;
+      case TR::java_lang_Long_longValue:
+         className = "Ljava/lang/Long;";
+         break;
+      case TR::java_lang_Float_floatValue:
+         className = "Ljava/lang/Float;";
+         break;
+      case TR::java_lang_Double_doubleValue:
+         className = "Ljava/lang/Double;";
+         break;
+      default:
+         TR_ASSERT_FATAL(false, "unrecognized unboxing method");
+      }
+   TR_OpaqueClassBlock * fieldClass = generator->fej9()->getClassFromSignature(className, strlen(className), generator->method());
 
+   // It's possible that the class is null because of a failed lookup.
+   // See eclipse-openj9/openj9#9416
```

Comment:
May I suggest that you expand a little bit on the reference to issue #9416?  Projects are sometimes forced to move from one source control system to another, and references like this can be left dangling.  The discussion in that issue is also rather long.